tox.ini

```diff
@@ -23,6 +23,8 @@ commands =
 
 [testenv:coverage{,-html}{,-fail}]
 depends = py{,38,39,310,311,312}
+# coverage should run on oldest supported Python
+base_python = py38,py39,py310,py311
```

Comment:
Clarifying the interpretation here - The behavior in CI definitely seems to be using the py38 environment; but does this definition mean it will prefer the py38 environment, but if you don't have one, it will use py39, and so on? The tox documentation isn't clear about this (to me, anyway), and it's difficult to test.

Reply:
Yeah....everything about `tox` just feels a little cryptic. Nonetheless, `tox` supports a whole "discovery" algorithm for finding versions of Python; depending on `PATH` is mostly working for me, though.

This setting means use the first available version of Python in this list.

The easiest way I found to test it is to make sure Python 3.8 or Python 3.9 is in `PATH` and start `tox` from a virtual env using a different Python version. You should see `tox` create an environment using 3.8...sometimes adding `-v` to `tox` makes things clearer....or even more confusing...

Finally, CI is was using Python 3.8 even before this because that's what's installed for that CI job and the default choice for `which python`.